nomenklatura/index/tantivy_index.py

```diff
@@ -148,3 +151,28 @@ def match(self, entity: CE) -> List[Tuple[Identifier, float]]:
             # Value of type "Document" is not indexable
             results.append((Identifier.get(doc["entity_id"][0]), score))  # type: ignore
         return results
+
+    def pairs(self) -> List[Tuple[Pair, float]]:
+        """
+        Compare all matchable entities in the index and return pairs in order of
+        similarity.
+        """
+        pairs: Dict[Pair, float] = defaultdict(lambda: 1.0)
+
+        for entity in self.view.entities():
+            if not entity.schema.matchable or entity.id is None:
+                continue
+
+            query = self.entity_query(entity)
+            ident = Identifier(entity.id)
+            for score, address in self.searcher.search(query, self.max_candidates).hits:
+                if score < self.threshold:
+                    break
+                # Value of type "Document" is not indexable
+                doc = self.searcher.doc(address)
+                other_ident = Identifier(doc["entity_id"][0])  # type: ignore
+                if ident == other_ident:
+                    continue
+                pair = (max(ident, other_ident), min(ident, other_ident))
+                pairs[pair] *= score
```

Comment:
What if this got the number of pairs to be generated as an argument and just threw away low-scored candidates (which are not going to be used even though they're bigger than threshold) every now and then?

Reply:
Actually that doesn't work because it would need to know the resolver....